lib/spack/spack/cmd/python.py

```diff
@@ -48,10 +51,47 @@ def python(parser, args, unknown_args):
     if unknown_args:
         tty.die("Unknown arguments:", " ".join(unknown_args))
 
+    # If arguments or a command are provided, just use python
+    if args.python_args or args.python_command:
+        args.python_interpreter = "python"
+
+    # Run user choose of interpreter
+    if args.python_interpreter == "ipython":
```

Comment:
If the user specifies `-i` and doesn't select `ipython` I think Spack should fail with an error message (in other words, I think there should be an `else` at the end of this branch that raises an exception if no expected identifier was provided).

Also, is there another interpreter besides `IPython` that could be used here (i.e. one available as a library)? If not, it might make sense to make `-i` a flag.

Reply:
Argparse should handle that. The choices are `-i ipython` or `-i python` (the default).

Reply:
@scheibelp there are indeed other interpreters - jython and bpython as quick examples. In practice I'm not sure if it's likely a user would request these specific ones over ipython or python, but since it could very likely be a possibility, I think we should maintain the `-i` to maintain consistency of the UI and then allow for it. For example, if spack some day has some kind of interaction with a notebook (to quickly load and do some kind of quick data science thing with package metadata), the "interpreter" could very easily be something like jupyter.